src/lib/dnssd/tests/TestIncrementalResolve.cpp

```diff
@@ -0,0 +1,427 @@
+    NL_TEST_ASSERT(inSuite, name.Get() != AsSerializedQName(kTestCommissionerNode));
+    NL_TEST_ASSERT(inSuite, name.Get() == AsSerializedQName(kTestCommissionableNode));
+
+    {
+        // setting to a too long value should reset it
+        uint8_t largeBuffer[256];
+        for (unsigned i = 0; i < 20; i++)
+        {
+            memcpy(largeBuffer + i * 8, "\07abcd123", 8);
+            largeBuffer[(i + 1) * 8] = 0;
+
+            // Up to 64 bytes supported to copy
+            if (i < 7)
+            {
+                NL_TEST_ASSERT_LOOP(inSuite, i, name.Set(AsSerializedQName(largeBuffer)) == CHIP_NO_ERROR);
+                NL_TEST_ASSERT_LOOP(inSuite, i, name.Get() == AsSerializedQName(largeBuffer));
+                NL_TEST_ASSERT_LOOP(inSuite, i, name.Get() != AsSerializedQName(kTestOperationalName));
+            }
+            else
+            {
+                NL_TEST_ASSERT_LOOP(inSuite, i, name.Set(AsSerializedQName(largeBuffer)) == CHIP_ERROR_NO_MEMORY);
+                NL_TEST_ASSERT_LOOP(inSuite, i, !name.Get().Next());
+            }
+        }
+        NL_TEST_ASSERT(inSuite, name.Set(AsSerializedQName(largeBuffer)) == CHIP_ERROR_NO_MEMORY);
+        NL_TEST_ASSERT(inSuite, !name.Get().Next());
+    }
+}
+
+void TestCreation(nlTestSuite * inSuite, void * inContext)
+{
+    IncrementalResolver resolver;
+
+    NL_TEST_ASSERT(inSuite, !resolver.IsActive());
+    NL_TEST_ASSERT(inSuite, !resolver.IsActiveCommissionParse());
+    NL_TEST_ASSERT(inSuite, !resolver.IsActiveOperationalParse());
+    NL_TEST_ASSERT(inSuite,
+                   resolver.GetRequiredInformation().HasOnly(IncrementalResolver::RequiredInformation::kSrvInitialization));
+}
+
+void TestStartOperational(nlTestSuite * inSuite, void * inContext)
+{
+    IncrementalResolver resolver;
+
+    NL_TEST_ASSERT(inSuite, !resolver.IsActive());
+
+    SrvRecord srvRecord;
+    PreloadSrvRecord(inSuite, srvRecord);
+
+    NL_TEST_ASSERT(inSuite, resolver.InitializeParsing(AsSerializedQName(kTestOperationalName), srvRecord) == CHIP_NO_ERROR);
+
+    NL_TEST_ASSERT(inSuite, resolver.IsActive());
+    NL_TEST_ASSERT(inSuite, !resolver.IsActiveCommissionParse());
+    NL_TEST_ASSERT(inSuite, resolver.IsActiveOperationalParse());
+    NL_TEST_ASSERT(inSuite, resolver.GetRequiredInformation().HasOnly(IncrementalResolver::RequiredInformation::kIpAddress));
+    NL_TEST_ASSERT(inSuite, resolver.GetServerName() == AsSerializedQName(kTestServerName));
+}
+
+void TestStartCommissionable(nlTestSuite * inSuite, void * inContext)
+{
+    IncrementalResolver resolver;
+
+    NL_TEST_ASSERT(inSuite, !resolver.IsActive());
+
+    SrvRecord srvRecord;
+    PreloadSrvRecord(inSuite, srvRecord);
+
+    NL_TEST_ASSERT(inSuite, resolver.InitializeParsing(AsSerializedQName(kTestCommissionableNode), srvRecord) == CHIP_NO_ERROR);
+
+    NL_TEST_ASSERT(inSuite, resolver.IsActive());
+    NL_TEST_ASSERT(inSuite, resolver.IsActiveCommissionParse());
```

Comment:
Why is it called "CommissionParse" when it's "Commissionable" (which is a spec term). Suggest may rename to `IsActiveAndParsingCommissionable` or such

Reply:
Sure